### PR TITLE
 Sync init.sh up with linux-distro-cmds.sh in systemready-es-sr-template and fix IR automation test issue with debug_dump.nsh

### PR DESCRIPTION
**Pull request by sunnywang-arm**
Two changes:  
1. Sync init.sh up with linux-distro-cmds.sh in systemready-es-sr-template
https://gitlab.arm.com/systemready/systemready-es-sr-template/-/blob/main/os-logs/linux-distro-cmds.sh
So that we don't need to ask partner to run linux-distro-cmds.sh again to collect the commands' output that haven't been covered by init.sh
The commented-out commands means the ACS Linux still haven't supported them or has issue with them. 

2. Fixed the issue that the uefi_dump folder gets created in an unexpected file system
The issue was reported by IR ACS users on https://github.com/ARM-software/arm-systemready/pull/187#discussion_r1639306891
Basically, the change is just moving the "endfor" from line 88 to line 34 so that it won't change the FS number for the case where there is no acs_result folder.

Signed-off-by: Sunny Wang <sunny.wang@arm.com>

### CODE REVIEW

File: common/ramdisk/init.sh [hunk 1]
```diff
@@ -104,17 +104,39 @@ if [ $ADDITIONAL_CMD_OPTION != "noacs" ]; then
...
+ # timedatectl > /mnt/acs_results/linux_dump/timedatectl.txt
+ hwclock > /mnt/acs_results/linux_dump/hwclock.log
  efibootmgr > /mnt/acs_results/linux_dump/efibootmgr.log
+ efibootmgr -t 20 > /mnt/acs_results/linux_dump/efibootmgr-t-20.log
+ efibootmgr -t 5 > /mnt/acs_results/linux_dump/efibootmgr-t-5.log
+ efibootmgr -c > /mnt/acs_results/linux_dump/efibootmgr-c.txt
+ ifconfig > /mnt/acs_results/linux_dump/ifconfig.log
+ ip addr show > /mnt/acs_results/linux_dump/ip-addr-show.log
+ ping -c 5 www.arm.com > /mnt/acs_results/linux_dump/ping-c-5-www-arm-com.log
+ # acpidump > /mnt/acs_results/linux_dump/acpi.log
```
Comment (edhay) [inline common/ramdisk/init.sh:131]: Query: Why are the below commands commented out?
Is it intentional?

**Review by edhay — APPROVED** [review]

**Comment by sunnywang-arm** [comment]
Hi @edhay and @chetan-rathore 

I wasn't aware that this pull request hadn't been merged, so I accidentally updated the pull request to include one more change which is a fix for an issue with IR ACS automation.  Could you review and merge this pull request? 

Cc @ownia

File: common/ramdisk/init.sh (hunk 1 above)
Comment (chetan-rathore) [inline common/ramdisk/init.sh:131]: Hi @sunnywang-arm, 

In the #189, request is made to add few acpi related command dump, but the commands requested to be added are commented in this PR.
Is there any particular reason for these commands to be commented.

Comment (sunnywang-arm) [reply common/ramdisk/init.sh:131]: Hi @chetan-rathore 
The acpi related commands require acpica-tools which hasn't been added/ported to ACS Linux. If we run these commands, we'll just see the error messages. In other words, we'll uncomment these commands once acpica-tools gets added to ACS Linux.
